File: artifacts/libzahl-1.0/decompose/src/zswap.c
void zswap(z_t a, z_t b)
{
  z_t t;
  *t = *a;
  *a = *b;
  *b = *t;
}


/*** DEPENDENCIES:
typedef struct 
{
  int sign;
  size_t used;
  size_t alloced;
  zahl_char_t *chars;
} z_t[1]
----------------------------
***/

File: artifacts/libzahl-1.0/translate_new/src/zswap.rs
use crate::*;

pub fn zswap(a: &mut z_t, b: &mut z_t) {
    // Create a temporary Z to hold the value
    let mut t = Z {
        sign: 0,
        used: 0,
        alloced: 0,
        chars: None,
    };
    
    // Swap the values using safe Rust operations
    std::mem::swap(&mut a[0], &mut t);
    std::mem::swap(&mut b[0], &mut a[0]);
    std::mem::swap(&mut t, &mut b[0]);
}
